infra/modules/service/execution-access.tf

```diff
@@ -0,0 +1,31 @@
+#-----------------
+# ECS Exec Access
+# See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html
+#-----------------
+resource "aws_iam_policy" "ecs_exec" {
+  name        = "${var.service_name}-ecs-exec"
+  description = "A policy to run ECS Exec"
+  policy      = data.aws_iam_policy_document.ecs_exec.json
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_exec" {
+  count      = var.enable_service_execution ? 1 : 0
+  role       = aws_iam_role.app_service.name
+  policy_arn = aws_iam_policy.ecs_exec.arn
+}
+
+data "aws_iam_policy_document" "ecs_exec" {
```

Comment:
nit: can we move the policy document block up to be next to the policy resource and have the policy attachment at the bottom